packages/types/src/client.ts

```diff
@@ -176,52 +176,52 @@ export interface Client<O extends ClientOptions = ClientOptions> {
    * Register a callback for whenever a span is started.
    * Receives the span as argument.
    */
-  on(hook: 'spanStart', callback: (span: Span) => void): void;
+  on(hook: 'spanStart', callback: (span: Span) => void): () => void;
```

Comment:
Let's add a comment to the JSDocs about what's returned from the methods

Reply:
I addressed your comments, I see there're some random failures happening between commits, not sure if they're related...